### PR TITLE
Migration: Add checkbox to toggle selection for all users on user import screen

### LINKED ISSUE #88402: Migration Flow: Add "Select all users" toggle to import users screen
Add "Select all users" toggle to user import screen.

Design: DASWajWXHUUjF4pu1yo4fI-fi-580%3A541

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes https://github.com/Automattic/wp-calypso/issues/88402

## Proposed Changes

* Adds "Select all users" checkbox toggle to Migration flow user import screen.

CleanShot 2024-03-12 at 00 03 35@2x

## Testing Instructions

* Use live link or checkout branch and go through migration flow until you reach import user screen (`sites?hosting-flow=true`)
* Verify the checkbox is working correctly.
* Verify user invitation is working correctly.
* Try to break or find issues.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx [hunk 1]
```diff
@@ -109,9 +109,32 @@ const ImportUsers = ( { site, onSubmit }: Props ) => {
...
+
+	const renderToggleAllUsers = () => {
+		const allUsersChecked = usersList.every( ( user ) => user.checked );
+		const toggleAllUsers = () => {
+			const updatedUsersList = usersList.map( ( user ) => {
+				return { ...user, checked: ! allUsersChecked };
+			} );
+			setUsersList( updatedUsersList );
+			setCheckedUsersNumber( updatedUsersList.filter( ( x ) => x.checked )?.length );
+		};
```
Comment (escapemanuele) [inline client/blocks/importer/wordpress/import-everything/import-users/index.tsx:126]: Nice use of logic

**Review by escapemanuele — APPROVED** [review]
Wonderful 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11601993

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11601993).

**Pull request merged**